source/NVDAHelper.py

```diff
@@ -398,7 +398,14 @@ def nvdaControllerInternal_inputLangChangeNotify(threadID,hkl,layoutString):
 def nvdaControllerInternal_typedCharacterNotify(threadID,ch):
 	focus=api.getFocusObject()
 	if focus.windowClassName!="ConsoleWindowClass":
-		eventHandler.queueEvent("typedCharacter",focus,ch=ch)
+		# Manually queue a call to executeEvent rather than using queueEvent,

```

Comment:
Have you considered either one of the following: 

1. Adding an additional kwarg to eventHandler.queueEvent that allows bypassing the pending events system. If that kwarg is semi private (e.g. by prefixing it with an underscore), it won't hurt anyone and it simplifies the API.
2. Tune the behavior of the garbage collector, i.e. by disabling the automatic collection and running the garbage collector manually at certain intervals, within the core pump, etc. Alternatively, we could temporarily disable collection in situations that could cause deadlocks.

Reply:
https://docs.python.org/3/library/gc.html